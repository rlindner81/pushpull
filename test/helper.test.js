"use strict";

const { assert, ordinal, escapeRegExp } = require("../src/helper");

test("assert", () => {
  expect(() => assert(false, "help")).toThrow("help");
});

test("!assert", () => {
  expect(() => assert(true, "help")).not.toThrow();
});

test("ordinal", () => {
  const numbers = [...Array(40).keys()].map((a) => a - 20);
  expect(numbers.map(ordinal)).toMatchSnapshot();
});

test("escapeRegExp", () => {
  const readableAsciiChars = [...Array(128).keys()]
    .slice(32)
    .map((i) => String.fromCharCode(i))
    .join("");
  expect(escapeRegExp(readableAsciiChars)).toMatchSnapshot();
});
