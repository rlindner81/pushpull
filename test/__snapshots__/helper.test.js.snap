// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escapeRegExp 1`] = `" !"#\\$%&'\\(\\)\\*\\+,-\\./0123456789:;<=>\\?@ABCDEFGHIJKLMNOPQRSTUVWXYZ\\[\\\\\\]\\^_\`abcdefghijklmnopqrstuvwxyz\\{\\|\\}~"`;

exports[`ordinal 1`] = `
[
  "-20th",
  "-19th",
  "-18th",
  "-17th",
  "-16th",
  "-15th",
  "-14th",
  "-13th",
  "-12th",
  "-11th",
  "-10th",
  "-9th",
  "-8th",
  "-7th",
  "-6th",
  "-5th",
  "-4th",
  "-3rd",
  "-2nd",
  "-1st",
  "0th",
  "1st",
  "2nd",
  "3rd",
  "4th",
  "5th",
  "6th",
  "7th",
  "8th",
  "9th",
  "10th",
  "11th",
  "12th",
  "13th",
  "14th",
  "15th",
  "16th",
  "17th",
  "18th",
  "19th",
]
`;
